File: frontend/src/types/featurePermission.ts
import type { UserRole } from './auth';

export interface FeaturePermission {
  id: number;
  featureKey: string;
  featureName: string;
  category: string;
  description?: string;
  allowedRoles: UserRole[];
  enabled: boolean;
  sortOrder: number;
}

export interface FeaturePermissionUpdateRequest {
  allowedRoles: UserRole[];
  enabled?: boolean;
}

File: backend/src/main/java/com/aisay/manga/entity/FeaturePermission.java
package com.aisay.manga.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("feature_permissions")
public class FeaturePermission {

    private Long id;

    private String featureKey;

    private String featureName;

    private String category;

    private String description;

    private String allowedRoles;

    private Boolean enabled;

    private Integer sortOrder;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}
